src/internal/connector/exchange/service_iterators.go

```diff
@@ -125,8 +125,22 @@ func filterContainersAndFillCollections(
 			newDelta.Reset)
 
 		collections[cID] = &edc
-		edc.added = append(edc.added, added...)
-		edc.removed = append(edc.removed, removed...)
+
+		// This results in "last one wins" if there's duplicate entries for an ID
```

Comment:
Are the delta records ordered? I know the `[]api.DeltaResult` is ordered but asking about the graph delta records we built that slice from?

Reply:
it's unclear to me if they are or not. The documentation for OneDrive seems to imply they are and the last one should be used. I found some mentions of getting records multiple times and the comments there were to merge the changes. Since we're just dealing with whether or not an item exists it's a bit fuzzier.

I guess we could always keep additions and handle the failure when fetching the item data if we want? I don't think we have code for that at the moment though

Reply:
Going to go ahead and merge this. As we discussed offline - we should probably just switch to "delete wins" because we won't see an item with the same ID reappear after delete